### DISCUSSION TITLE
Adding user-agent to twirp client

### DISCUSSION THREAD

**Discussion by prayansh** [twirphp/twirp#199] [Q&A]
Hi! 
I am using the auto-generated twirp clients and am looking for a way to easily attach the User-Agent header to all outgoing requests. Here is the way I am currently doing it:

```golang
// UserAgentClient is a wrapper which ensures that a User-Agent header is
// provided with all requests.
type UserAgentClient struct {
	userAgent string
	base      rpc.HTTPClient
}

// NewUserAgentClient wraps the base HTTP client with one that will always set
// the User-Agent header.
func NewUserAgentClient(base rpc.HTTPClient, userAgent string) *UserAgentClient {
	return &UserAgentClient{
		base:      base,
		userAgent: userAgent,
	}
}

// Do sets the User-Agent header and delegates the rest of the request handling
// to the wrapped HTTP client.
func (c *UserAgentClient) Do(req *http.Request) (*http.Response, error) {
	req.Header.Set("User-Agent", c.userAgent)
	return c.base.Do(req)
}

func main() {
	baseHttpClient := NewUserAgentClient(
                 &http.Client{},
		"my-user-agent",
	)
	reportsClient := generated.NewXXXProtobufClient("service-url", baseHttpClient)
}
```
but it feels very heavy-handed, since if i wanted to do this across repositories I would need to replicate the `UserAgentClient` code in each repo

I was hoping there are simpler ways to do it via Interceptors / Client options that others are using. Any suggestions for the same?

**Comment by sagikazarmark** [comment]
Attaching HTTP headers to outgoing requests is documented here: https://twirphp.github.io/docs/beyond-the-basics/custom-http-headers

TwirPHP uses the PSR HTTP client interface and I'm fairly certain most clients already support setting a user agent header.
